e2e/healthcheck_update_test.go

```diff
@@ -0,0 +1,98 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/exec"
+	"google.golang.org/api/networkservices/v1"
+)
+
+func TestHealthcheckUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	meshURL, meshKey := ensureMesh(ctx, t, "hc-update-test-mesh")
+	t.Cleanup(func() {
+		err := theCloud.Meshes().Delete(ctx, meshKey)
+		t.Logf("theCloud.Meshes().Delete(ctx, %s): %v", meshKey, err)
+	})
+
+	graphBuilder := rgraph.NewBuilder()
+	negID, err := buildNEG(graphBuilder, "neg-hc-update-test", zone)
+	if err != nil {
+		t.Fatalf("buildNEG(_, neg-test, %s) = (_, %v), want (_, nil)", zone, err)
+	}
+	t.Cleanup(func() {
+		err := theCloud.NetworkEndpointGroups().Delete(ctx, negID.Key)
+		t.Logf("theCloud.NetworkEndpointGroups().Delete(ctx, %s): %v", negID.Key, err)
+	})
+
+	hcID, err := buildHealthCheck(graphBuilder, "hc-update-test", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc-update-test, 15) = (_, %v), want (_, nil)", err)
+	}
+	t.Cleanup(func() {
```

Comment:
health check, backend service and tcproutes (all resources that are managed by rGraph) cleanup can be in one function after executor run

Reply:
Done.